lib/queue_classic/worker.rb

```diff
@@ -161,5 +163,15 @@ def setup_queues(adapter, queue, queues, top_bound)
       end
     end
 
+    def clone_args(job)
+      job[:args].map do |arg|
+        begin
+          arg.clone
+        rescue TypeError
```

Comment:
It's covering `Fixnum`, which throws a `TypeError` when you try to clone it (there are a bunch of tests that cover this case already, although I could add another explicit one). I can't find another general way to find out if an object can be safely cloned—`frozen?` delivers `true` for `Fixnum`s in Ruby >=2.0.0 but `false` in <=1.9.3, and `respond_to?` returns `true` for `Fixnum` even though it blows up. The other option I guess is to specifically check for `Fixnum` or `Float`, which would probably be safe because it's serialized from JSON.